src/org/thoughtcrime/securesms/InstantOnboardingActivity.java

```diff
@@ -0,0 +1,438 @@
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.text.util.Linkify;
+import android.util.Log;
+import android.view.MenuItem;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import androidx.loader.app.LoaderManager;
+
+import com.b44t.messenger.DcContext;
+import com.b44t.messenger.DcEvent;
+import com.b44t.messenger.DcLot;
+import com.bumptech.glide.load.engine.DiskCacheStrategy;
+import com.bumptech.glide.request.target.CustomTarget;
+import com.bumptech.glide.request.transition.Transition;
+import com.google.zxing.integration.android.IntentIntegrator;
+import com.google.zxing.integration.android.IntentResult;
+import com.soundcloud.android.crop.Crop;
+
+import org.thoughtcrime.securesms.components.AvatarSelector;
+import org.thoughtcrime.securesms.connect.DcEventCenter;
+import org.thoughtcrime.securesms.connect.DcHelper;
+import org.thoughtcrime.securesms.contacts.avatars.ResourceContactPhoto;
+import org.thoughtcrime.securesms.mms.AttachmentManager;
+import org.thoughtcrime.securesms.mms.GlideApp;
+import org.thoughtcrime.securesms.permissions.Permissions;
+import org.thoughtcrime.securesms.profiles.AvatarHelper;
+import org.thoughtcrime.securesms.profiles.ProfileMediaConstraints;
+import org.thoughtcrime.securesms.qr.RegistrationQrActivity;
+import org.thoughtcrime.securesms.scribbles.ScribbleActivity;
+import org.thoughtcrime.securesms.util.Prefs;
+import org.thoughtcrime.securesms.util.views.ProgressDialog;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.SecureRandom;
+import java.util.Objects;
+
+public class InstantOnboardingActivity extends BaseActionBarActivity implements DcEventCenter.DcEventDelegate {
+
+  private static final String TAG = InstantOnboardingActivity.class.getSimpleName();
+  private static final String DCACCOUNT = "dcaccount";
+  private static final String DCLOGIN = "dclogin";
+  private static final String INSTANCES_URL = "https://delta.chat/chatmail";
+  private static final String DEF_CHATMAIL_HOST = "nine.testrun.org";
+
+  public static final String QR_ACCOUNT_EXTRA = "qr_account_extra";
+  private static final int REQUEST_CODE_AVATAR = 1;
+
+  private ImageView avatar;
+  private EditText name;
+  private TextView privacyPolicyBtn;
+  private Button signUpBtn;
+
+  private boolean avatarChanged;
+  private boolean imageLoaded;
+  private String providerHost;
+  private String providerQrData;
+  private boolean isDcLogin;
+
+  private AttachmentManager attachmentManager;
+  private Bitmap avatarBmp = null;
+  private ProgressDialog progressDialog = null;
```

Comment:
by default, Java initialises instance variables to `null`  - we we do not set them elsewhere manually, we shouln't do here either:

```suggestion
  private Bitmap avatarBmp;
  private ProgressDialog progressDialog;
```

(did not test :)